[docs] add split view docs (#41559)
expo/expo: docs/pages/versions/unversioned/sdk/router-split-view.mdx

      <PasscodeCard title="Security" param="security" />
      <PasscodeCard title="Deleted" param="deleted" />
    </SafeAreaView>
  );
}

const passkeys = ['Github', 'Google', 'Facebook', 'Twitter', 'Apple', 'Microsoft', 'Amazon'];
const security = ['Admin1234', 'Root'];

const all = [...passkeys, ...security];

function PasswordElementList() {
  const params = useGlobalSearchParams();
  const data = (() => {
    switch (params.type) {
      case 'all':
      case undefined:
        return all;
      case 'passkeys':
        return passkeys;
      case 'security':
        return security;
      default:
        return [];
    }
  })();
  return (
    <ScrollView contentInsetAdjustmentBehavior="automatic" style={{ backgroundColor: undefined }}>
      {data.map(item => (
        <PasswordElement key={item} title={item} />
      ))}
    </ScrollView>
  );
}

function PasscodeCard({ param, title }: { param: string; title: string }) {
  const params = useGlobalSearchParams();
  const isActive = params.type === param;
  return (
    <Link
      href={`/${param}/`}
      disabled={isActive}
      style={[
        style.passcodeCard,
        {

>           backgroundColor: isActive ? PlatformColor('systemBlue') : PlatformColor('systemGray6'),

Reviewer: Use new `Color` API

---
title: Router Split View
description: An Expo Router submodule that provides native split view layout.
sourceCodeUrl: 'https://github.com/expo/expo/tree/main/packages/expo-router'
packageName: 'expo-router'
platforms: ['ios']
isAlpha: true
---

import { BookOpen02Icon } from '@expo/styleguide-icons/outline/BookOpen02Icon';

import APISection from '~/components/plugins/APISection';
import { BoxLink } from '~/ui/components/BoxLink';
import { Collapsible } from '~/ui/components/Collapsible';
import { ConfigPluginExample } from '~/ui/components/ConfigSection';
import { ContentSpotlight } from '~/ui/components/ContentSpotlight';
import { FileTree } from '~/ui/components/FileTree';

> > **important** SplitView is currently in [alpha](/more/release-statuses/#alpha) and available in [canary releases](/versions/latest/#using-pre-release-versions) of `expo-router`. It requires configuration in **app.json** and is not available in Expo Go.

Reviewer 2: ```suggestion
> **important** SplitView is currently in [alpha](/more/release-statuses/#alpha) and available in [canary releases](/versions/latest/#using-pre-release-versions) of `expo-router`. You need to use a [development build](/develop/development-builds/introduction) since it is unavailable in Expo Go.
```

In the callout, lets mention what people need to do, that is to use development builds.

`expo-router/unstable-split-view` is a submodule of `expo-router` and exports components to build split view layouts using [platform-native system split views](https://developer.apple.com/design/human-interface-guidelines/split-views).

> See the [Expo Router](./router) reference for more information about the file-based routing library for native and web app.

> ### Platform support

Reviewer 2: ```suggestion
## Platform support
```

Since this is the first section on the page with a header, lets make it an H2. Sorry for not seeing this before.

To use `expo-router/unstable-split-view` in your project, you need to install `expo-router` in your project. Follow the instructions from Expo Router's installation guide:

<BoxLink
  title="Install Expo Router"
  description="Learn how to install Expo Router in your project."
  href="/router/installation/"
  Icon={BookOpen02Icon}
/>

## Configuration in app config

> To enable Split View support, add the `unstable_splitView` property to the `expo-router` plugin configuration in your app config file (app.json or app.config.js):

Reviewer 2: ```suggestion
To enable Split View support, add the `unstable_splitView` property to the `expo-router` plugin configuration in your [app config](/workflow/configuration/) file:
```

Let's link the app config doc here since `app.config.ts` is also a valid format and adding all formats is too verbose as per our writing style guide.

  alt="A screenshot of an iPad app using split view with two columns."
  src="/static/images/expo-router/split-view-two-column-layout.png"
  className="max-h-[400px]"
/>

```tsx app/_layout.tsx
import { Link } from 'expo-router';
import { SplitView } from 'expo-router/unstable-split-view';
import { Text, Pressable } from 'react-native';
import { SafeAreaView } from 'react-native-screens/experimental';

export default function Layout() {
  return (
    <SplitView>
      <SplitView.Column>
        <SafeAreaView style={{ flex: 1 }}>
          <Link href="/inbox">
            <Pressable style={{ padding: 16 }}>
              <Text>Inbox</Text>
            </Pressable>
          </Link>
          <Link href="/sent">
            <Pressable style={{ padding: 16 }}>
              <Text>Sent</Text>
            </Pressable>
          </Link>
        </SafeAreaView>
      </SplitView.Column>
    </SplitView>
  );
}
```

### Three-column layout

A sidebar with supporting column and main content:

<ContentSpotlight
  alt="A screenshot of an iPad app using split view with three columns."
  src="/static/images/expo-router/split-view-three-column-layout.png"
  className="max-h-[400px]"
/>

> ```tsx

Reviewer 2: ````suggestion
```tsx app/_layout.tsx
````

      <Text style={{ color: Color.ios.label, fontSize: 24, fontWeight: 'bold' }}>
        Nothing is selected
      </Text>
    </View>
  );
}
```

```tsx app/[type]/[id].tsx
import { useLocalSearchParams } from 'expo-router';
import { Text, View } from 'react-native';

export default function Id() {
  const { id } = useLocalSearchParams();

  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text>ID: {id}</Text>
    </View>
  );
}
```

## Platform support

Split View is only available on iOS. On other platforms, the `SplitView` component automatically falls back to rendering as a standard `Slot` navigator, ensuring your app works across all platforms without conditional code.

## Known limitations

<Collapsible summary="Cannot be nested">

There can only be one `SplitView` in the navigation hierarchy. Attempting to nest split views will result in an error.

</Collapsible>

<Collapsible summary="Cannot be used inside other navigators">

`SplitView` cannot be used inside another navigator (except for `Slot`). It must be used at the root layout level.

</Collapsible>

<Collapsible summary="Requires prebuild">

> After enabling split view in **app.json**, you must run `npx expo prebuild` to apply the native configuration. The feature won't work in Expo Go.

Reviewer 2: ```suggestion
After enabling split view in app config, you must run `npx expo prebuild` to apply the native configuration. The feature won't work in Expo Go.
```
